### PR TITLE
dygraph Prim code gen （node.cc)

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types
<!--  Others 

### PR changes
<!--  Others

### Describe
增加动态图node.cc 中对composite字段解析的相关生成代码， 目前按照flag_prim_enable判断同时生成组合算子和融合算子分支，后续去除分支，每个反向只会有一种算子执行路径，且使用组合算子后，不在有创建更高阶的node实例代码

### CODE REVIEW

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py [hunk 1]
```diff
@@ -2328,6 +2354,9 @@ def GenerateNodeDefinition(
             var_str += f"\n{indent}  output_str += output_{new_name}_str; "
 
         log_str = AFTER_LOG_PRINT_TEMPLATE.format(var_str)
+        # TODO Ruting modify in the future
```
Comment (JiabinYang) [inline paddle/fluid/eager/auto_code_generator/generator/eager_gen.py]: TODO with wrong format

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py [hunk 2]
```diff
@@ -1826,16 +1833,19 @@ def GenerateHigherOrderNodeCreationCode(self):
         is_invoke_forward_api = IsInvokeForwardApi(
             self.grad_api_contents, self.forward_apis_dict
         )
+        is_composite_forward_api = False if self.composite_func_info == [] else True
```
Comment (JiabinYang) [inline paddle/fluid/eager/auto_code_generator/generator/eager_gen.py]: why composite forward api?

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py [hunk 3]
```diff
@@ -1826,16 +1833,19 @@ def GenerateHigherOrderNodeCreationCode(self):
...
             return (
                 has_higher_order_node,
                 is_invoke_forward_api,
+                is_composite_forward_api,
                 next_grad_node_creation_str,
                 next_grad_node_out_list,
                 next_node_generator.backward_forward_inputs_map,
             )
-        elif not is_invoke_forward_api:
+        elif not is_invoke_forward_api and not is_composite_forward_api:
```
Comment (JiabinYang) [inline paddle/fluid/eager/auto_code_generator/generator/eager_gen.py]: what if it hits else branch?

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py [hunk 4]
```diff
@@ -2225,6 +2241,16 @@ def GenerateNodeDefinition(
   {out_assign_str}}} else {{
   {indent}{autograd_api_out} api_output = paddle::experimental::{self.namespace}{self.grad_api_contents['invoke']};
   {out_assign_str}{indent}}}
+  """
+        elif is_composite_grad_api:
```
Comment (JiabinYang) [inline paddle/fluid/eager/auto_code_generator/generator/eager_gen.py:2256]: Leave TODO here to indicate strategy which will be used later, such as using composite only when we don't have backward kernel

Comment (JiabinYang) [reply paddle/fluid/eager/auto_code_generator/generator/eager_gen.py:2256]: This should be statically generated here

**Review by JiabinYang — COMMENTED** [review]
some comments

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py (hunk 3 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/eager/auto_code_generator/generator/eager_gen.py]: if is_invoke_farward_api or is_composite_grad_api next_grad_node_creation_str should be none, we will add this when delete Flags_prim_enabled
add TODO here

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py (hunk 2 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/eager/auto_code_generator/generator/eager_gen.py]: replace by composite grad api

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py (hunk 1 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/eager/auto_code_generator/generator/eager_gen.py]: modified

File: paddle/fluid/eager/auto_code_generator/generator/eager_gen.py (hunk 4 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/eager/auto_code_generator/generator/eager_gen.py:2256]: done

File: paddle/fluid/prim/api/manual/utils/CMakeLists.txt [hunk 5]
```diff
@@ -2,3 +2,8 @@ cc_library(
   static_utils
   SRCS static_utils.cc
   DEPS proto_desc operator static_global_utils)
+
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/manual/utils/CMakeLists.txt:5]: Avoid this ON INFER and NO PYTHON

**Review by JiabinYang — COMMENTED** [review]
one comment

File: paddle/fluid/prim/tests/CMakeLists.txt [hunk 6]
```diff
@@ -15,32 +15,13 @@ set(prim_eager_deps
     grad_tensor_holder
     custom_operator_node)
 
-set(prim_generated_deps final_dygraph_function final_dygraph_node
-                        dygraph_function dygraph_node)
-
-cc_test_old(
+set(prim_generated_deps final_dygraph_function final_dygraph_node dygraph_function
```
Comment (JiabinYang) [inline paddle/fluid/prim/tests/CMakeLists.txt]: why change this?

File: paddle/fluid/prim/api/generated/prim_gen.py [hunk 7]
```diff
@@ -0,0 +1,490 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/generated/prim_gen.py]: maybe no need to push these code in this pr?

File: paddle/fluid/eager/auto_code_generator/generator/CMakeLists.txt [hunk 8]
```diff
@@ -47,8 +47,8 @@ add_custom_target(
           ${forwards_cc_path}
   COMMAND ${CMAKE_COMMAND} -E copy_if_different ${tmp_forwards_h_path}
           ${forwards_h_path}
-  #   COMMAND ${CMAKE_COMMAND} -E copy_if_different ${tmp_nodes_cc_path}
-  #           ${nodes_cc_path}
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${tmp_nodes_cc_path}
```
Comment (JiabinYang) [inline paddle/fluid/eager/auto_code_generator/generator/CMakeLists.txt:50]: remove additional generated files

**Review by JiabinYang — COMMENTED** [review]
some comments

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**